### PR TITLE
Established app/apply for registration page

### PR DESCRIPTION
**Pull request by acauf564**

### CODE REVIEW

**Comment by Kapocsi** [comment]
I added a "feature flag", not really because I am too lazy to setup feature flags
properly on vercel. But currently there is a edge config that we check to see if
this page should be deployed served. Once this page is done, we can either remove
the check or change the state of flag.

For now I will send the extra thing you need in the .env.local file in our discord

**Review by Kapocsi — APPROVED** [review]

File: lib/auth/signInWithGithub.ts [hunk 1]
```diff
@@ -19,5 +19,6 @@ export default async function signInWithGithub() {
     return;
   }
 
+  console.log(data.url);
```
Comment (Kapocsi) [inline lib/auth/signInWithGithub.ts:22]: I don't think this is needed

File: lib/auth/signInWithGithub.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export default async function signInWithGithub() {
   const { data, error } = await supabase.auth.signInWithOAuth({
     provider: "github",
     options: {
-      redirectTo: getCallbackUrl(),
+      redirectTo: `${getCallbackUrl()}?next=apply`,
```
Comment (Kapocsi) [inline lib/auth/signInWithGithub.ts:13]: This is usually safe; I think we are only working with data we trust, but we should really avoid stuff like this. Unthinkingly concatenating can lead to all sorts of problems and is thus an antipattern. 

I would take a look at the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL_API) and use its [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)  interface to more safely to this.

### example in my node repl

```
Welcome to Node.js v22.3.0.
Type ".help" for more information.
> url_str = "https://example.com"
'https://example.com'
> url = new URL(url_str)
URL {
  href: 'https://example.com/',
  origin: 'https://example.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'example.com',
  hostname: 'example.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> url.searchParams.set("next", "apply")
undefined
> url.toString()
'https://example.com/?next=apply'
```

File: app/auth/callback/route.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export async function GET(request: Request) {
 
     const { error } = await supabase.auth.exchangeCodeForSession(code);
     if (!error) {
-      return NextResponse.redirect(`${origin}${next}`);
+      return NextResponse.redirect(`${origin}/${next}`);
```
Comment (Kapocsi) [inline app/auth/callback/route.ts:33]: Same thing here as below, we should be using the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL_API). I know this is not your code originally, but its stuff like this that makes it easier to do routing logic idiomatically. Without need to manged trailing/leading slashes:

> See how all these have the same href:
```
Welcome to Node.js v22.3.0.
Type ".help" for more information.
> new URL("apply", "https://example.com").href
'https://example.com/apply'
> new URL("/apply", "https://example.com").href
'https://example.com/apply'
> new URL("/apply/", "https://example.com").href
'https://example.com/apply/'
> new URL("apply/", "https://example.com").href
'https://example.com/apply/'
```

**Review by Kapocsi — CHANGES_REQUESTED** [review]

File: lib/auth/signInWithGithub.ts (hunk 1 above)
Comment (acauf564) [reply lib/auth/signInWithGithub.ts:22]: My b - didn't mean to leave that here :)

**Review by Kapocsi — APPROVED** [review]

**Pull request merged**